From 23f8c5e41b0d54285acd45edf6ec50ee65f428f3 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Tue, 9 Jul 2019 19:39:47 +0800
Subject: [PATCH 36/41] HACK: glamor: Use copy bail funcs for 64x64 images for
 freerdp

Freerdp would split screen into 64x64 images.
For small images, the glamor's performance is worse.

But preparing accessing requires buffer sync, so let's only hack the
64x64 images for freerdp case.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 glamor/glamor_copy.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/glamor/glamor_copy.c b/glamor/glamor_copy.c
index 1ab2be6c0..82ae20dd1 100644
--- a/glamor/glamor_copy.c
+++ b/glamor/glamor_copy.c
@@ -701,6 +701,12 @@ glamor_copy_gl(DrawablePtr src,
     glamor_pixmap_private *src_priv = glamor_get_pixmap_private(src_pixmap);
     glamor_pixmap_private *dst_priv = glamor_get_pixmap_private(dst_pixmap);
 
+#ifdef GLAMOR_HAS_GBM_MAP
+    /* HACK for freerdp */
+    if (nbox == 1 && (box->x2 - box->x1) == 64 && (box->y2 - box->y1) == 64)
+        return FALSE;
+#endif
+
     if (GLAMOR_PIXMAP_PRIV_HAS_FBO(dst_priv)) {
         if (GLAMOR_PIXMAP_PRIV_HAS_FBO(src_priv)) {
             if (glamor_copy_needs_temp(src, dst, box, nbox, dx, dy))
-- 
2.11.0

